### PR TITLE
Feature: implement `tracing` logs as a feature

### PR DESCRIPTION
**Pull request by Samuel-B-D**
the request tracing is integrated into the handler code instead of as a middleware to be able to wrap the FileMiddleware log span within the request span regardless of middleware order.

This PR also fix controller macro's handler span so that tracing span within the macro output have proper line numbers, instead of pointing to the top #[controller] macro.

### CODE REVIEW

**Comment by Samuel-B-D** [comment]
Eh, clippy don't pass ( https://github.com/richerarc/saphir/actions/runs/3132727251/jobs/5085367660 ) but I'm curious as to why it's not caught on my end :thinking: 
image

**Comment by Samuel-B-D** [comment]
> Eh, clippy don't pass ( https://github.com/richerarc/saphir/actions/runs/3132727251/jobs/5085367660 ) but I'm curious as to why it's not caught on my end thinking image

Fixed

File: saphir/src/utils.rs [hunk 1]
```diff
@@ -558,6 +558,10 @@ where
     serde_urlencoded::from_str::<T>(query_str)
 }
 
+// #[cfg(feature = "tracing-instrument")]
```
Comment (LucFauvel) [inline saphir/src/utils.rs:561]: commented code?

**Review by LucFauvel — CHANGES_REQUESTED** [review]

File: saphir/src/utils.rs (hunk 1 above)
Comment (Samuel-B-D) [reply saphir/src/utils.rs:561]: Oops, I cleaned up my test stuff afterward but left some behind it seems.
Removing this shortly

**Review by LucFauvel — APPROVED** [review]

**Pull request merged**